frontend/app/services/subscription-service.server.ts

```diff
@@ -242,41 +277,111 @@ function createSubscriptionService() {
 
-    const response = await fetch(url, {
-      method: 'PUT',
+    if (!response.ok) {
+      log.error('%j', {
+        message: 'Failed find data',
+        status: response.status,
+        statusText: response.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
+
+      throw new Error(`Failed to find data. Status: ${response.status}, Status Text: ${response.statusText}`);
+    }
+
+    const user = await response.json();
+    const userParsed = userSchema.parse(user);
+
+    const userSubscriptionsURL = userParsed._links.subscriptions.href;
+    const subscriptionsResponse = await fetch(userSubscriptionsURL, {
+      method: 'POST',
       headers: {
         'Content-Type': 'application/json',
       },
       body: JSON.stringify(subscription),
     });
 
-    instrumentationService.countHttpStatus('http.client.cdcp-api.subscriptions.posts', response.status);
+    instrumentationService.countHttpStatus('http.client.cdcp-api.subscriptions.create', subscriptionsResponse.status);
 
-    if (response.status === 204) {
-      return null;
+    if (!subscriptionsResponse.ok) {
+      log.error('%j', {
+        message: 'Failed to create data',
+        status: subscriptionsResponse.status,
+        statusText: subscriptionsResponse.statusText,
+        url: url,
+        responseBody: await subscriptionsResponse.text(),
+      });
+
+      throw new Error(`Failed to create data. Status: ${subscriptionsResponse.status}, Status Text: ${subscriptionsResponse.statusText}`);
     }
+  }
+
+  async function updateSubscription(userId: string, subscription: SubscriptionInfo) {
+    const auditService = getAuditService();
+    const instrumentationService = getInstrumentationService();
+    auditService.audit('alert-subscription.update', { userId });
+
+    const url = new URL(`https://api.cdcp.example.com/api/v1/users/${userId}`);
+
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    instrumentationService.countHttpStatus('http.client.cdcp-api.subscriptions.posts', response.status);
 
     if (!response.ok) {
       log.error('%j', {
-        message: 'Failed to update data',
+        message: 'Failed find data',
         status: response.status,
         statusText: response.statusText,
         url: url,
         responseBody: await response.text(),
       });
 
-      throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
+      throw new Error(`Failed to find data. Status: ${response.status}, Status Text: ${response.statusText}`);
+    }
+
+    const user = await response.json();
+    const userParsed = userSchema.parse(user);
+
+    const subscriptionsUrl = userParsed._links.subscriptions.href;
+    const updateSubscriptionUrl = `${subscriptionsUrl}/${subscription.id}`;
+    const updateSubscriptionResponse = await fetch(updateSubscriptionUrl, {
```

Comment:
I think this should use the new `instrumentedFetch()` function: <https://github.com/DTS-STN/canadian-dental-care-plan/blob/main/frontend/app/utils/fetch-utils.server.ts#L55>

Reply:
updated all fetch